GitPullRequest.Services/GitRepositoryFactory.cs

```diff
@@ -0,0 +1,25 @@
+﻿using LibGit2Sharp;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace GitPullRequest.Services
+{
+    public static class GitRepositoryFactory
+    {
+        public static GitRepository Create(IRepository repo, string remoteName)
+        {
+            var uri = new Uri(repo.Network.Remotes[remoteName].Url);
+
+            if (uri.Host.Equals("github.com", StringComparison.OrdinalIgnoreCase))
+            {
+                return new GitHubRepository(repo, remoteName);
+            }
+            else if (uri.Host.EndsWith(".visualstudio.com", StringComparison.OrdinalIgnoreCase))
```

Comment:
I'm using an old school `.visualstudio.com` address as well, but I imagine we'll also want to support the newer `dev.azure.com` domain. I wonder if anything else will need to be updated to support `dev.azure.com` based URLs?

Reply:
I was going to add `.azure.com` to this just in case but looked at [this docs page](https://docs.microsoft.com/en-us/azure/devops/repos/git/clone?view=vsts&tabs=visual-studio) and every URL was just `.visualstudio.com`, so I left it.. I should have checked though and I can access the VS repo, for exampe, through `devdiv.visualstudio.com` and `dev.azure.com/devdiv` so I guess they're interchangeable.

Reply:
> I should have checked though and I can access the VS repo, for exampe, through devdiv.visualstudio.com and dev.azure.com/devdiv so I guess they're interchangeable.

Does this mean both project URLs and clone URLs can use the new `dev.azure.com/{your organization}/{your project}` format?

Reply:
It looks that way externally at least. Navigating to a repo via each URL shows me a clone URL that uses the same format I navigated with